packages/tools/src/GoogleSearch/GoogleSearchTool.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Modelflow AI package.
+ *
+ * (c) Johannes Wachter <johannes@sulu.io>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ModelflowAi\Tools\GoogleSearch;
+
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+readonly class GoogleSearchTool
+{
+    public function __construct(
+        private HttpClientInterface $client,
+        private string $googleCustomSearchEngineId,
+        private string $googleCustomSearchEngineApiKey,
+        private string $url = 'https://www.googleapis.com/customsearch/v1',
+    ) {
+    }
+
+    /**
+     * Use this tool in the following circumstances:
```

Comment:
@Prokyonn This does the trick 🎉  with that the chat uses the function quite effective but only if you provide the URL